packages/react-components/lib/tasks/create-task.tsx

```diff
@@ -355,26 +351,68 @@ export function CreateTaskForm({
 
   const isScreenHeightLessThan800 = useMediaQuery('(max-height:800px)');
 
-  const [favoriteTaskBuffer, setFavoriteTaskBuffer] = React.useState<TaskFavorite>(
-    defaultFavoriteTask(),
-  );
+  // Note that we are not checking if the number of supported tasks is larger
+  // than 0, this will cause the dashboard to fail when the create task form is
+  // opened. This is intentional as it is a misconfiguration and will require
+  // the build-time configuration to be fixed.
+  let allTasksToDisplayAreValid = true;
+  const validTasks: TaskDefinition[] = [];
+  let defaultTaskDescription: string | TaskDescription | null = null;
+  let defaultTaskRequest: TaskRequest | null = null;
+  tasksToDisplay.forEach((supportedTask: TaskDefinition, index: number) => {
+    const definitionId = supportedTask.taskDefinitionId;
+    const desc = getDefaultTaskDescription(definitionId);
+    const req = getDefaultTaskRequest(definitionId);
+
+    if (desc === undefined) {
+      console.error(`Failed to retrieve task description for definition ID: [${definitionId}]`);
+      allTasksToDisplayAreValid = false;
+    }
+    if (req === null) {
+      console.error(`Failed to create task request for definition ID: [${definitionId}]`);
+      allTasksToDisplayAreValid = false;
+    }
+    if (desc && req) {
+      validTasks.push(supportedTask);
+
+      if (!defaultTaskDescription && !defaultTaskRequest) {
+        defaultTaskDescription = desc;
+        defaultTaskRequest = req;
+      }
+    }
+  });
+
+  if (!defaultTaskDescription || !defaultTaskRequest) {
+    // We should never reach this state unless a misconfiguration happened.
+    const err = Error('Default task could not be generated, this might be a configuration error');
+    onFail && onFail(err, []);
+    console.error(err.message);
+    throw new TypeError(err.message);
```

Comment:
This is technically not a type error, just throw the `err` created earlier.

Reply:
https://github.com/open-rmf/rmf-web/pull/925/commits/ba6be3062714389593027cdac565d0e593027c1b